app/Allocator/Squawk/General/CcamsSquawkAllocator.php

```diff
@@ -3,102 +3,23 @@
 namespace App\Allocator\Squawk\General;
 
 use App\Allocator\Squawk\AbstractSquawkAllocator;
-use App\Allocator\Squawk\SquawkAssignmentInterface;
-use App\Models\Squawk\Ccams\CcamsSquawkAssignment;
 use App\Models\Squawk\Ccams\CcamsSquawkRange;
-use App\Services\NetworkDataService;
-use Illuminate\Support\Collection;
-use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Eloquent\Builder;
 
-class CcamsSquawkAllocator extends AbstractSquawkAllocator implements GeneralSquawkAllocatorInterface
+class CcamsSquawkAllocator extends AbstractSquawkAllocator
 {
-    public function allocate(string $callsign, array $details): ?SquawkAssignmentInterface
+    protected function getOrderedSquawkRangesQuery(array $details): Builder
     {
-        $assignment = null;
-        DB::transaction(
-            function () use (&$assignment, $callsign) {
-                CcamsSquawkRange::all()->shuffle()->each(
-                    function (CcamsSquawkRange $range) use (&$assignment, $callsign) {
-                        $squawks = $range->getAllSquawksInRange();
-                        $possibleSquawks = $squawks->diff(
-                            CcamsSquawkAssignment::whereIn('code', $squawks)->pluck('code')->all()
-                        );
-
-                        if ($possibleSquawks->isEmpty()) {
-                            return true;
-                        }
-
-                        $assignment = $this->assignSquawkFromAvailableCodes(
-                            $callsign,
-                            $possibleSquawks->shuffle()
-                        );
-                        return false;
-                    }
-                );
-            }
-        );
-
-        return $assignment;
-    }
-
-    public function delete(string $callsign): bool
-    {
-        return CcamsSquawkAssignment::destroy($callsign);
+        return CcamsSquawkRange::query();
     }
 
-    public function fetch(string $callsign): ?SquawkAssignmentInterface
+    protected function canAllocateSquawk(array $details): bool
     {
-        return CcamsSquawkAssignment::find($callsign);
+        return true;
     }
 
-    public function assignToCallsign(string $code, string $callsign): ?SquawkAssignmentInterface
+    protected function getAssignmentType(): string
```

Comment:
Does this need to be a function or can it just be a property that is overridden from the abstract class?

Reply:
Think it could be either - I like it functional as its then quite easy to follow the trail!